### PR TITLE
Package deployment fixes

### PR DESCRIPTION
**Pull request by mihails-strasuns-sociomantic**
No idea if it will work :)

### CODE REVIEW

**Comment by mihails-strasuns-sociomantic** [comment]
Seems to work, though rather ugly.

File: .travis.yml [hunk 1]
```diff
@@ -52,24 +52,28 @@ jobs:
     include:
         # Test matrix
         - <<: *test-matrix
-          env: DMD=1.081.* F=production
+          env: DMD=1.081.* F=production CACHE_ID=$TRAVIS_BUILD_ID
```
Comment (leandro-lucarella-sociomantic) [inline .travis.yml]: I guess making it global will break it, right?

Comment (mihails-strasuns-sociomantic) [reply .travis.yml]: Not sure. https://docs.travis-ci.com/user/caching/#Caches-and-build-matrices implies they simply hash `env` value as a single string and use it as key - in that case globals won't matter. But doc says "all visible env vars" which would mean globals included.

Comment (leandro-lucarella-sociomantic) [reply .travis.yml]: I guess it would make sense.

**Comment by leandro-lucarella-sociomantic** [comment]
I'm a bit ambivalent about this one, on one side it looks quite ugly and intrusive, but on the other side I can't see a much better option (the only alternative I see is renaming `.deb` packages to remove the version info (and hope that this is not a problem with bintray).

**Comment by mihails-strasuns-sociomantic** [comment]
Updated with less noisy version

**Comment by mihails-strasuns-sociomantic** [comment]
Also added fix to missing D2 packages FYI @gavin-norman-sociomantic

**Comment by leandro-lucarella-sociomantic** [comment]
FYI @joseph-wakeling-sociomantic

**Comment by mihails-strasuns-sociomantic** [comment]
@gavin-norman-sociomantic I should probably rebase it against currently deployed point release branch to fix lack of current packages. Which one would it be?

**Comment by gavin-norman-sociomantic** [comment]
v1.12.x

**Comment by mihails-strasuns-sociomantic** [comment]
Changed base

**Comment by gavin-norman-sociomantic** [comment]
Should the commit title "Try using build ID as cache key" actually be "Use build ID as cache key"?

**Comment by mihails-strasuns-sociomantic** [comment]
I am not 100% confident it will actually work :D

**Comment by gavin-norman-sociomantic** [comment]
Oh still? Haha. Ok let's merge.

**Pull request merged**